### ISSUE TITLE
api: Use pgxscan to simplify library access

### ISSUE THREAD

**Issue by mitjat** [oasisprotocol/oasis-indexer#206]
We can make our SQL querying code shorter _and_ safer, at the cost of an extra dependency.

Every API handler that reads several rows from the DB (i.e. most of them) currently looks like this, given `rows := db.Query(...)`:

```go
	ts := RuntimeTransactionList{
		Transactions: []RuntimeTransaction{},
	}
	for rows.Next() {
		var t RuntimeTransaction
		if err := rows.Scan(
			&t.Round,
			&t.Index,
			&t.Hash,
			&t.EthHash,
			&t.Raw,
			&t.ResultRaw,
		); err != nil {
			c.logger.Info("row scan failed",
				"request_id", ctx.Value(RequestIDContextKey),
				"err", err.Error(),
			)
			return nil, common.ErrStorageError
		}

		ts.Transactions = append(ts.Transactions, t)
	}

	return &ts, nil
```

Using https://github.com/randallmlough/pgxscan (or https://github.com/georgysavva/scany - more established), we can reduce this to
```go
var ts []RuntimeTransaction{}
if err := pgxscan.NewScanner(rows).Scan(&ts); err != nil { 
			c.logger.Info("row scan failed",
				"request_id", ctx.Value(RequestIDContextKey),
				"err", err.Error(),
			)
			return nil, common.ErrStorageError
}
return RuntimeTransactionList{ Transactions: ts }, nil
```

Not only is this more concise, it's safer because it matches each row's fields to struct fields by name, not by position.

Also, why do we log these errors at `Info` level only? Also, the whole `if err :=` block with error logging could be abstracted into a helper func.

**Comment by pro-wh** [comment]
> Also, why do we log these errors at `Info` level only?

was it that scanning is what actually raises things like 'there is no block at that height' kind of errors?

**Comment by mitjat** [comment]
> was it that scanning is what actually raises things like 'there is no block at that height' kind of errors?

No, this query should work just fine (= produce no errors) when it returns 0 rows. But I think you might be onto something -- before Andy introduced arg parsing as a separate layer, input args were inserted into the SQL more directly, with no semantics checks. So this was likely assigned a low log level because external users were able to trigger errors here with nonsensical inputs. Now that those are caught earlier on in the API layer, we should raise the log level again.
